### ISSUE TITLE
Mention that Binary Classification is also called Logistic Regression

### ISSUE THREAD

**Issue by gcolonna12** [d2l-ai/d2l-en#2642]
Mention that Binary Classification is also called Logistic Regression. And also answer why even if it's called "regression", it's actually a classification problem

**Comment by muzammil5539** [comment]
> Mention that Binary Classification is also called Logistic Regression. And also answer why even if it's called "regression", it's actually a classification problem


In binary classification,logistic regression uses the sigmoid function to map predictions to probabilities:

\[
P(y=1\mid x)=\frac{1}{1+e^{-(w\cdot x+b)}}
\]


Example:
Suppose we have a trained model with:

• Weights\(w=[2]\)

• Bias\(b=-3\)

For a data point\(x=1\):
\[
z=w\cdot x+b=2\cdot 1-3=-1
\]
\[
P(y=1\mid x=1)=\frac{1}{1+e^{-(-1)}}\approx 0.269
\]

Interpretation:The probability of belonging to ClassB(1)is approximately26.9%,so we predictClassA(0).

For\(x=3\):
\[
z=2\cdot 3-3=3
\]
\[
P(y=1\mid x=3)=\frac{1}{1+e^{-3}}\approx 0.953
\]

Interpretation:The probability is95.3%,so we predictClassB(1).


Summary:

• If\(P(y=1\mid x)\geq 0.5\),predictClassB(1).

• If\(P(y=1\mid x)<0.5\),predictClassA(0).

**Comment by muzammil5539** [comment]
> Mention that Binary Classification is also called Logistic Regression. And also answer why even if it's called "regression", it's actually a classification problem


properly explain this concept with example:
In **binary classification** we can use **0** to represent classA and **1** to represent classB.

and if we take logistic Regression formula:

logistic Regression = 1/(1+e^-x)

it's value range from 0 to 1 i.e [0,1]

and in what way we represent binary classes classA=0 and classB = 1

but in logistic regression value range in between 0 and 1 so it tells how much x value belong to a binary class


for example:

if x = -3

y = 1/(1+e^-(-3)) = approx(0.047)

which is close to 0 means classA

if  x = 3
y = 1 /(1+e^-3) = approx(0.953)

which is close to 1 means classB and so on..
try with different values to get the hang of it... 
if y=< 0.5 classA and y> 0.5 classB

I hope I relayed my point

**Comment by muzammil5539** [comment]
> > Mention that Binary Classification is also called Logistic Regression. And also answer why even if it's called "regression", it's actually a classification problem
> 
> properly explain this concept with example: In **binary classification** we can use **0** to represent classA and **1** to represent classB.
> 
> and if we take logistic Regression formula:
> 
> logistic Regression = 1/(1+e^-x)
> 
> it's value range from 0 to 1 i.e [0,1]
> 
> and in what way we represent binary classes classA=0 and classB = 1
> 
> but in logistic regression value range in between 0 and 1 so it tells how much x value belong to a binary class
> 
> for example:
> 
> if x = -3
> 
> y = 1/(1+e^-(-3)) = approx(0.047)
> 
> which is close to 0 means classA
> 
> if x = 3 y = 1 /(1+e^-3) = approx(0.953)
> 
> which is close to 1 means classB and so on.. try with different values to get the hang of it... if y=< 0.5 classA and y> 0.5 classB
> 
> I hope I relayed my point


for clarity purpose x here does not refer to input x but it refers to the logits produce by the system the system and logits doubt is cleared here: https://github.com/d2l-ai/d2l-en/issues/2645
